packages/cli/src/commands/add.ts

```diff
@@ -160,8 +159,7 @@ export default class AddCommand extends Command {
 
     try {
       if (isLocalHost) throw Error; // Triggers user prompting without waiting for Etherscan lookup to fail
-
-      contractName = await loadContractNameForAddress(network, address);
+      contractName ||= await loadContractNameForAddress(network, address);
```

Comment:
While we are at it, could you move `contractName` uniqueness check from L87-L94 to below this try-catch so we can capture name loaded from etherscan too